libs/components/forms/src/lib/modules/checkbox/checkbox-group.component.html

```diff
@@ -6,52 +6,69 @@
     headingText && formGroup?.errors ? errorId : undefined
   "
 >
-  <legend
-    class="sky-control-label"
-    [ngClass]="{
-      'sky-screen-reader-only': headingHidden,
-      'sky-margin-stacked-xs': !!hintText,
-      'sky-margin-stacked-sm': !hintText
-    }"
-  >
-    <span
-      class="sky-margin-inline-xs"
-      [ngClass]="{
-        'sky-control-label-required': required
-      }"
-    >
-      @if (headingLevel === 3) {
-        <h3 [class]="headingClass">
-          {{ headingText }}
-        </h3>
-      } @else if (headingLevel === 4) {
-        <h4 [class]="headingClass">
-          {{ headingText }}
-        </h4>
-      } @else if (headingLevel === 5) {
-        <h5 [class]="headingClass">
-          {{ headingText }}
-        </h5>
-      } @else {
-        <span [class]="'sky-heading-text ' + headingClass">
-          {{ headingText }}
-        </span>
+  <legend class="sky-screen-reader-only">
```

Comment:
This feels like a major rewrite to the component when all it needs is a harness. Can you provide some context to this change to help me understand why it's needed for this story?

Reply:
reverted